plugins/guests/darwin/cap/change_host_name.rb

```diff
@@ -6,7 +6,7 @@ def self.change_host_name(machine, name)
           if !machine.communicate.test("hostname -f | grep '^#{name}$' || hostname -s | grep '^#{name}$'")
             machine.communicate.sudo("scutil --set ComputerName #{name}")
             machine.communicate.sudo("scutil --set HostName #{name}")
-            machine.communicate.sudo("scutil --set LocalHostName #{name}")
+            machine.communicate.sudo("scutil --set LocalHostName #{name.split('.')[0]}")
```

Comment:
Ouch. This is a nasty bug. 

I think it might be unexpected if we just take the first `.` without at least telling the user. Can we do something like this instead?

```ruby
safe_hostname = name
if name.include?(".")
  safe_hostname = name.split(".")[0]
  Vagrant.log.warn("Detected a dot in hostname `#{name}' for OS X. Using `#{safe_hostname}' because OS X does not support dots in hostname."
end
```

And then pass `safe_hostname` to the `--set` command. This way the user knows we are doing something different for them.